### PR TITLE
Add confirmation input widget

### PR DESCRIPTION
**Pull request by vinothvino42**
In this PR, I added a confirmation input widget with schema
There are three types of Input Type - (default, rounded, underline)

**EDL**
```yaml
ConfirmationInput:
  id: confirmationInputId
  styles:
    length: 5
    enableCursor: true
    autoComplete: true
    fieldType: default
    inputType: number
    defaultFieldBorderColor: 0xFF969BE9
    activeFieldBorderColor: 0xFF969BE9
    filledFieldBorderColor: black
    defaultFieldBackgroundColor: 0xFFE4DFFF
    activeFieldBackgroundColor: 0xFF969BE9
    filledFieldBackgroundColor: 0xFF7B63F8
    cursorColor: 0xFFE4DFFF
    textStyle:
      color: white
      fontSize: 20
      fontWeight: bold
  onChange: |
    //@code
    onChangeTextId.text = 'Entered OTP is  ' + this.text;
    console.log("OnChange called");
  onComplete: |
    //@code
    onCompleteTextId.text = 'Submitted OTP is  ' + this.text;
    console.log("OnCompleted called");
```

### CODE REVIEW

**Comment by vusters** [comment]
- Call it "onComplete" instead of "onSubmit" to convey the user has completed the field. There is no submission here. 
- Need a boolean flag "autoComplete". True (default) will trigger onComplete when the user has typed the full length, False will trigger onComplete when the user has typed the full length and hit Enter.

File: lib/widget/confirmation_input.dart [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        defaultFieldBackgroundColor:
+            controller.defaultFieldBackgroundColor ?? Colors.transparent,
+        activeFieldBackgroundColor:
+            controller.activeFieldBackgroundColor ?? Colors.transparent,
+        filledFieldBackgroundColor:
+            controller.filledFieldBackgroundColor ?? Colors.transparent,
+        filledFieldBorderColor:
+            controller.filledFieldBorderColor ?? Colors.transparent,
+      ),
+      maxLength: controller.length ?? 4,
```
Comment (vusters) [inline lib/widget/confirmation_input.dart:115]: don't assume 4 here. If it's a required field, expose it.

**Comment by vusters** [comment]
Need schema changes

File: lib/widget/confirmation_input.dart (hunk 1 above)
Comment (vinothvino42) [reply lib/widget/confirmation_input.dart:115]: It's not a required field. So I length as a late property and in the setters, I set it as a 4 for the fallback value.

**Comment by vinothvino42** [comment]
@vusters I updated the ConfirmationInput widget

Now you can have the following
1. inputType - Keyboard input types like text, number, email, etc
2. autoComplete - True (default) will trigger onComplete when the user specified length and has typed the full length, False will trigger onComplete when the user hit Enter.
3. Based on the inputType, the input formatters also included (This doesn't allow the user not to enter or copy paste any other values)

I changed the onSubmit to onComplete and also I set the length as the late property with fallback value of 4.

NOTE: I just forked the otp_pin_field package and modified most of the properties which are not exposed to the user. So now our ensemble's otp_pin_field points to my forked repo (develop branch)

**Review by vusters — COMMENTED** [review]
please put this forked repo on ensemble

**Review by vusters — APPROVED** [review]
you can merge once you put the forked repo on ensemble

File: lib/widget/confirmation_input.dart [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  }
+
+  @override
+  Map<String, Function> setters() {
+    return {
+      'fieldType': (input) =>
+          _controller.fieldType = Utils.optionalString(input),
+      'enableCursor': (newValue) =>
+          _controller.enableCursor = Utils.getBool(newValue, fallback: true),
+      'length': (newValue) => _controller.length = Utils.optionalInt(newValue),
```
[the remark this answers is not in the document]
Comment (vinothvino42) [reply lib/widget/confirmation_input.dart]: The default value of length is 4 internally in the `otp_pin_field` package itself. So it's an optional one only

**Pull request merged**